graphql_compiler/compiler/blocks.py

```diff
@@ -412,3 +423,11 @@ class Unfold(MarkerBlock):
     def validate(self):
         """Unfold blocks are always valid in isolation."""
         pass
+
+
+class EndOptional(MarkerBlock):
```

Comment:
Since there's no `StartOptional` marker block, describe how optionals are considered started for the purposes of this marker.